src/proxy/cpsubsystem/BaseCPProxy.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/** @ignore *//** */
+
+import * as Promise from 'bluebird';
+import {HazelcastClient} from '../../HazelcastClient';
+import {ClientMessage} from '../../protocol/ClientMessage';
+import {UnsupportedOperationError} from '../../core';
+
+/**
+ * Common super class for any CP Subsystem proxy.
+ * @internal
+ */
+export abstract class BaseCPProxy {
```

Comment:
This seems very similar to `BaseProxy`, is there a special use case of this?

Reply:
It's similar, but different, as it has no `destroy` implementation, throws on `getPartitionKey` call, has a different implementation of `encodeInvokeOnRandomTarget` method, and has additional properties. As CP DSes have a different manager class (and a different overall behavior), I don't see any good reason to use `BaseProxy` class here.